packages/commonwealth/server/models/group.ts

```diff
@@ -0,0 +1,67 @@
+import type * as Sequelize from 'sequelize';
+import { DataTypes } from 'sequelize';
+import { ChainAttributes } from './chain';
+import { ModelInstance, ModelStatic } from './types';
+import { Requirement } from '../util/requirementsModule/requirementsTypes';
+
+export type GroupMetadata = {
+  name: string;
+  description: string;
+  required_requirements: number;
+  membership_ttl: number;
+};
+
+export type GroupAttributes = {
+  id: number;
+  chain_id: string;
+  metadata: GroupMetadata;
+  requirements: Requirement[];
+
+  created_at?: Date;
+  updated_at?: Date;
+
+  // associations
+  Chain?: ChainAttributes;
+};
+
+export type GroupInstance = ModelInstance<GroupAttributes>;
+export type GroupModelStatic = ModelStatic<GroupInstance>;
+
+export default (
+  sequelize: Sequelize.Sequelize,
+  dataTypes: typeof DataTypes
+): GroupModelStatic => {
+  const Group = <GroupModelStatic>sequelize.define(
+    'Group',
+    {
+      id: { type: dataTypes.INTEGER, autoIncrement: true, primaryKey: true },
+      chain_id: { type: dataTypes.STRING, allowNull: false },
+      metadata: { type: dataTypes.JSON, allowNull: false },
```

Comment:
We should consider JSON vs JSONB types for the database object -- take a look at the docs and see which one might be ideal for this use case?

Reply:
Basically JSONB is more flexible and performant for querying/updating specific fields, and allows modifying data in place without replacing the whole document. Also supports indexing and full-text search.

The only notable downside for JSONB is that it has slower insert performance due to binary conversion overhead. Seeing as groups will likely not be written to as frequently as memberships, this might be negligible.

The planned implementation don't perform any complex queries on the JSON– the data is treated as a blob and fed into the requirements validator logic. I figured we weren't losing anything by choosing JSON over JSONB.

As for the group update operation, I believe it's more of a "put" rather than a "patch"?

Reply:
Group update operation is a put if requirements change, patch otherwise... kind of a weird situation. We might want to use JSONB for storage space? Not sure though, I think JSON is fine for now.

Reply:
Ironically, JSONB uses more disk space, assuming that the payload doesn't have unnecessary whitespace.